samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnClusterResourceManager.java

```diff
@@ -192,7 +193,8 @@ public YarnClusterResourceManager(Config config, JobModelManager jobModelManager
         clusterManagerConfig.getNumCores(),
         samzaAppState,
         state,
-        amClient
+        amClient,
+        new JobConfig(config).getApplicationMasterHighAvailabilityEnabled()
```

Comment:
Ideally, you want to follow through the `OnContainersAllocated` code path to ensure that all the state is intact when updating AM with the previous set of running containers.

We also need to ensure both `YarnAppState` and `SamzaApplicationState` are up-to-date on top of the other internal state held by `YarnResourceClusterManager` and `ContainerAllocator` barring the state that is interim during the container assignment & starting.

Reply:
updated to invoke OnContainerStarted instead of OnContainersAllocated -- as that seemed to be the more appropriate one (setting job healthy etc). 
fixed the npe arising from not having previous attempt containers in the allocatedResources map